src/viam/sdk/module/service.cpp

```diff
@@ -233,21 +233,38 @@ std::shared_ptr<Resource> ModuleService::get_parent_resource_(const Name& name)
     return parent_->resource_by_name(name);
 }
 
-ModuleService::ModuleService(std::string addr)
-    : module_(std::make_unique<Module>(std::move(addr))), server_(std::make_unique<Server>()) {
+ModuleService::ModuleService(std::string addr) : ModuleService(std::move(addr), "unix:") {}
+
+ModuleService::ModuleService(std::string addr, std::string grpc_conn_protocol)
+    : module_(std::make_unique<Module>(std::move(addr))),
+      grpc_conn_protocol_(std::move(grpc_conn_protocol)),
+      server_(std::make_unique<Server>()) {
     impl_ = std::make_unique<ServiceImpl>(*this);
 }
 
+namespace {
+std::string get_protocol(int argc, char** argv) {
+    for (int i = 0; i < argc; ++i) {
+        if (strcmp(argv[i], "--tcp-mode") == 0) {
+            return "dns:";
```

Comment:
Curious why this is `dns:`. Are we expecting hostnames here? Are we allowing cross host between modules and RDK?